packages/gatsby/src/utils/webpack-config.js

```diff
@@ -0,0 +1,290 @@
+    `EvalDevToolModulePlugin`,
+    `CachePlugin`,
+    `ExtendedAPIPlugin`,
+    `ExternalsPlugin`,
+    `JsonpTemplatePlugin`,
+    `LibraryTemplatePlugin`,
+    `LoaderTargetPlugin`,
+    `MemoryOutputFileSystem`,
+    `ProgressPlugin`,
+    `SetVarMainTemplatePlugin`,
+    `UmdMainTemplatePlugin`,
+    `NoErrorsPlugin`,
+    `NoEmitOnErrorsPlugin`,
+    `NewWatchingPlugin`,
+    `EnvironmentPlugin`,
+    `DllPlugin`,
+    `DllReferencePlugin`,
+    `LoaderOptionsPlugin`,
+    `NamedModulesPlugin`,
+    `NamedChunksPlugin`,
+    `HashedModuleIdsPlugin`,
+    `ModuleFilenameHelpers`,
+  ].forEach(plugin => {
+    plugins[pluginName(plugin)] = (...args) => new webpack[plugin](...args)
+  })
+  ;[
+    `AggressiveMergingPlugin`,
+    `AggressiveSplittingPlugin`,
+    `CommonsChunkPlugin`,
+    `ChunkModuleIdRangePlugin`,
+    `DedupePlugin`,
+    `LimitChunkCountPlugin`,
+    `MinChunkSizePlugin`,
+    `OccurrenceOrderPlugin`,
+    //`UglifyJsPlugin`
+  ].forEach(plugin => {
+    plugins[pluginName(plugin)] = (...args) =>
+      new webpack.optimize[plugin](...args)
+  })
+
+  /**
+   * The webpack2 shim plugin for passing options to loaders. Sets
+   * the minize and debug options to `true` in production (used by various loaders)
+   */
+  plugins.loaderOptions = options =>
+    new webpack.LoaderOptionsPlugin({
+      options,
+      minimize: PRODUCTION,
+      debug: !PRODUCTION,
+    })
+
+  /**
+   * Minify javascript code without regard for IE8. Attempts
+   * to parallelize the work to save time. Generally only add in Production
+   */
+  plugins.uglify = () =>
+    new webpack.optimize.UglifyJsPlugin({
+      sourceMap: true,
+      compress: {
+        screw_ie8: true, // React doesn't support IE8
+        warnings: false,
+      },
+      mangle: {
+        screw_ie8: true,
+      },
+      output: {
+        comments: false,
+        screw_ie8: true,
+      },
+    })
+
+  /**
+   * Extracts css requires into a single file;
+   * includes some reasonable defaults
+   */
+  plugins.extractText = options =>
+    new ExtractTextPlugin({
+      allChunks: true,
+
+      // Useful when using css modules, it assert a dependency order for how
+      // css is concated together. AS it is the default heuristic for ordering
+      // is require() order, but that is rarely indicative of the proper cascade
+      // order for component css files, and will cause issues when using "helper"
+      // files containing a lot of small composable classes
+      ignoreOrder: true,
+
+      // The extracted css is not needed for stage other than `build-css`
+      // but we extract at all build stages to avoid:
+      //  - adding extra weight to the js bundle in `build-javascript`
+      //  - errors in `build-html` caused by style-loader assuming a DOM exists
+      //
+      // TODO: these extra passes are costly on time and can be
+      // fixed in more optimal ways
+      disable: stage === `develop` || stage === `develop-html`,
```

Comment:
using this option instead of conditional adding the plugin lets gatsby plugins, just worry about making loaders use `extract.Text` and not have to worry about making sure the plugin runs are the appropriate stages, which isn't intuitive.